modules/ebpf/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+find_package(LIBBPF)
+
+module_switch(ENABLE_EBPF "Enable ebpf module (requires ebpf toolchain)" LIBBPF_FOUND)
+
+if (NOT ENABLE_EBPF)
+  return()
+endif()
+
+add_custom_command(OUTPUT vmlinux.h
+                   COMMAND ${BPFTOOL} btf dump file /sys/kernel/btf/vmlinux format c >vmlinux.h)
+
+add_custom_command(OUTPUT random.skel.c
+                   COMMAND ${BPFTOOL} gen skeleton random.kern.o > random.skel.c
+		   DEPENDS random.kern.o)
+
+add_custom_command(OUTPUT random.kern.o
+                   COMMAND ${BPF_CC} ${BPF_CFLAGS} -c ${CMAKE_CURRENT_SOURCE_DIR}/random.kern.c -o random.kern.o
+                   DEPENDS random.kern.c vmlinux.h)
+
+add_custom_target(generate_ebpf_skeletons DEPENDS "random.skel.c")
+
+set(EBPF_SOURCES
+    ebpf-parser.h
+    ebpf-reuseport.h
+    ebpf-reuseport.c
+    ebpf-plugin.c
+    ebpf-parser.c
+)
+
+add_module(
+  TARGET ebpf
+  GRAMMAR ebpf-grammar
+  INCLUDES ${PROJECT_SOURCE_DIR}
+  SOURCES ${EBPF_SOURCES}
+)
```

Comment:
`DEPENDS` is missing, so the final so does not link to `libebpf.so`.
```suggestion
add_module(
  TARGET ebpf
  GRAMMAR ebpf-grammar
  INCLUDES ${PROJECT_SOURCE_DIR}
  DEPENDS ${LIBBPF_LIBRARIES}
  SOURCES ${EBPF_SOURCES}
)
```

before
```
$ readelf -a lib/syslog-ng/libebpf.so  | grep NEEDED                                                                                   
 0x0000000000000001 (NEEDED)             Shared library: [libsyslog-ng.so.4.1.1.68.gea5051f]                                                                                                                         
 0x0000000000000001 (NEEDED)             Shared library: [libeventlog.so]                                                                                                                                            
 0x0000000000000001 (NEEDED)             Shared library: [libglib-2.0.so.0]                                                                                                                                          
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
```
after
```
$ readelf -a lib/syslog-ng/libebpf.so  | grep NEEDED
 0x0000000000000001 (NEEDED)             Shared library: [libbpf.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libsyslog-ng.so.4.1.1.68.gea5051f.dirty]
 0x0000000000000001 (NEEDED)             Shared library: [libeventlog.so]
 0x0000000000000001 (NEEDED)             Shared library: [libglib-2.0.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
```